### DISCUSSION TITLE
Vitotrans 353

### DISCUSSION THREAD

**Discussion by wangnick** [open3e/open3e#120] [Q&A]
Dear all,

a few days ago we got our Vitola oil heating replaced by a Vitocal 250-A, Vitocell 120-E and Vitotrans 353.

This evening I succeded in scanning the external CAN bus of the Vitocal 250-A using a RasPi 3B and a cheap MCP2512 CAN module. So far, so good.

Now, the Vitotrans 353 also features a CAN bus connector (Stecker 91, L - GND - H). But the voltages on the L and H pins versus GND are ... strange. 

Has anyone ever tried to read out a Vitotrans 353 via the CAN bus?

My goal is to track vital parameters such as gross energy production and then net energy delivery to radiators and warm water taps, so as to keep an eye on the losses.

KR, Sebastian

**Comment by philippoo66** [comment]
Moin Sebastian!

es ist noch zu früh am Morgen für Englsisch... Ist da schon ein Abschlusswiderstand auf dem CAN Port? Wie sind denn die Pegel gegen CAN_GND? oder wogegen hast du gemessen? Zwischen -27 und +40 Volt bsw. wäre für einen TJA1015 ok (siehe [hier](https://community.viessmann.de/t5/Konnektivitaet/CAN-Bus-Home-Automation-E3-Generation-lokal-und-kostenlos/m-p/393725#M49870))

Beachte bitte auch, dass du die beiden Geräte besser nicht an ein- und denselben CAN Bus anschliesst, wenn sie nicht entsprechend konfiguriert sind (2 Main ECUs)...

Grüsse!
Phil

**Reply by wangnick** [reply]
Ok, ich bin etwas weiter gekommen. Zwar flattern die Pegel von CANL und CANH, gemessen gegen das GND-Potential des RasPi, immer noch reichlich, eine CAN-Kommunikation ist aber trotzdem möglich. Dabei sind CANL und CANH an der Vitotrans und am RasPi jeweils über 120Ω verbunden.

![CAN Vitotrans](https://github.com/user-attachments/assets/7120296f-fe72-4741-9b29-fcad1556ef23)

Vielleicht sollte ich am RasPi  MCP2515-Modul CANL und CANH lieber jeweils über 60Ω miteinander und dieses Mittenpotential über einen Kondensator mit dem RasPi-GND verbinden? Oder sogar den CAN GND der Vitotrans mit dem GND des RasPi verbinden? Der Raspi hängt ansonsten nur an einem Schaltnetzteil (ohne PE-Kontakt des Steckern) und über ein Ethernetkabel (mit ungeschirmten Steckern) an einem Switch.

Die Vitotrans 353 und ihre DIDs werden aber so schon durch `open3e_depictSystem` wie folgt gefunden:
```
(.venv) vito@vitopi:~/vitotrans $ open3e_depictSystem -c can0
read DID enums ...
3038 DIDs listed.
scan COB-IDs 0x680 to 0x6ff ...
ECU found: 0x6e7 : FRIWASTATION
1 responding COB-IDs found.
write devices.json ...
done.
scan 6e7 for DIDs 256 to 3500 ...
found 256:36:BusIdentification
found 377:16:ViessmannIdentificationNumber
found 580:8:SoftwareVersion
found 581:8:HardwareVersion
found 928:16:ElectronicTraceabilityNumber
found 961:2:SecurityAlgorithmNumber
found 962:8:BootLoaderVersion
found 964:1:ActiveDiagnosticSession
found 1494:8:OemProductVersion
9 DIDs found on 6e7.
write datapoints file Open3Edatapoints_6e7.py ...
done.

(.venv) vito@vitopi:~/vitotrans $ open3e --config devices.json -a
reading 0x6e7, 9 datapoints, please be patient...
256 {'BusAddress': 86, 'BusType': {'ID': 3, 'Text': 'CanExternal'}, 'DeviceProperty': {'ID': 15, 'Text': 'FRIWASTATION'}, 'DeviceFunction': {'ID': 0, 'Text': 'NOTHING'}, 'SW-Version': '88.500.2309.2296', 'HW-Version': '88.500.2301.0', 'VIN': '77############86'}
377 77############86
580 88.500.2309.2296
581 88.500.2301.0
928 77############86
961 0400
962 88.500.2309.2296
964 1
1494 88.500.2309.2296
```

Leider scheinen sich in dieser DID-Liste die wirklich wertvollen Daten (Temperatursensor S1, S2, S3, S4, S5, Volumenstromsensor) nicht zu finden. Bummer! Macht es eventuell Sinn, den gescannten Bereich zu erweitern, also von 0 bis 9999 zu suchen?

LG, Sebastian

**Reply by wangnick** [reply]
> Beachte bitte auch, dass du die beiden Geräte besser nicht an ein- und denselben CAN Bus anschliesst, wenn sie nicht entsprechend konfiguriert sind (2 Main ECUs)...

Ist denn die Vitotrans unter 0x6e7 eine "Main ECU"?

**Reply by philippoo66** [reply]
mit Verlaub gesagt - dein Oszillogramm sieht ziemlich schei... aus. Leider kann man nicht erkennen, ob der 'Dreck' im Gleichtakt ist, wovon ich aber ausgehe. Was war denn die Bezugsmasse bei der Messung? Das mit dem "T-Stück" ist keine gute Idee. Wenn der Raspi potentialfrei ist (wie du ja schreibst) verbinde die GNDs.

Die Did Liste sieht tatsächlich etwas mager aus. Du kannst ja mal noch den Bereich 3500 bis 4000 scannen, aber ich glaube nicht dass du da was findest. Die Viguide 2.19.4 ist glaubich immer noch die aktuellste, und bei der war bei 3293 Schluss.

Vielleicht muss die Transe erstmal entsprechend für den Betrieb am CAN Bus konfiguriert werden, damit sie mehr Datenpunkte offenbart. Nach Main ECU sieht das aber trotzdem nicht aus. Alles irgenwie seltsam...

**Comment by wangnick** [comment]
Der "Dreck" war aber in der Tat im Gleichtakt. Dabei war die Bezugsmasse wie gesagt GND des RasPi. Ich werde beizeiten noch einmal einen Versuch starten und den GND des RasPi mit dem CAN GND der Vitotrans verbinden. Zur Zeit bin ich allerdings mit der Datenerhebung und -analyse der Vitocal (und mein Kundendienstmonteur mit der Behebung der Installationsdefizite) mehr als beschäftigt ...
